course_catalog/permissions.py

```diff
@@ -22,3 +23,26 @@ def has_object_permission(self, request, view, obj):
                 or request.user == obj.author
             )
         return request.user == obj.author
+
+
+class HasUserListItemPermissions(BasePermission):
+    """Permission to view/modify UserListItems"""
+
+    def has_permission(self, request, view):
+        user_list = UserList.objects.get(
```

Comment:
Might not be important but this will cause a 500 error when a user views this API with the wrong id. For a 404 we would need to use `get_object_or_404` or some similar logic.

Reply:
Fixed this and added a test for it